src/server/pfs/server/api_server.go

```diff
@@ -47,19 +56,43 @@ func (a *apiServer) CreateRepo(ctx context.Context, request *pfs.CreateRepoReque
 func (a *apiServer) InspectRepo(ctx context.Context, request *pfs.InspectRepoRequest) (response *pfs.RepoInfo, retErr error) {
 	func() { a.Log(request, nil, nil, 0) }()
 	defer func(start time.Time) { a.Log(request, response, retErr, time.Since(start)) }(time.Now())
+	metrics.ReportUserAction(ctx, "InspectRepoStarted", nil)
+	defer func(start time.Time) {
+		if retErr == nil {
+			metrics.ReportUserAction(ctx, "InspectRepoFinished", time.Since(start).Seconds())
+		} else {
+			metrics.ReportUserAction(ctx, "InspectRepoErrored", retErr.Error())
+		}
+	}(time.Now())
```

Comment:
Can we abstract this into a function? I'm really not excited about having 10 lines of boilerplate at the top of every method.